python/src/equistore/operations/join.py

```diff
@@ -48,36 +53,38 @@ def join(tensor_maps: List[TensorMap], axis: str):
 
     if len(tensor_maps) == 1:
         return tensor_maps[0]
-
-    for ts_to_join in tensor_maps[1:]:
-        _check_maps(tensor_maps[0], ts_to_join, "join")
-
-    # Deduce if sample/property names are the same in all tensor_maps.
-    # If this is not the case we have to change unify the corresponding labels later.
-    if axis == "samples":
-        names_list = [tensor.sample_names for tensor in tensor_maps]
-    else:
-        names_list = [tensor.property_names for tensor in tensor_maps]
-
-    # We use functools to flatten a list of sublists::
-    #
-    #   [('a', 'b', 'c'), ('a', 'b')] -> ['a', 'b', 'c', 'a', 'b']
-    #
-    # A nested list with sublist of different shapes can not be handled by np.unique.
-    unique_names = np.unique(functools.reduce(operator.concat, names_list))
-
-    # Label names are unique: We can do an equal check only checking the lengths.
-    names_are_same = np.all(
-        len(unique_names) == np.array([len(names) for names in names_list])
-    )
-
-    # It's fine to loose metadata on the property axis, less so on the sample axis!
-    if axis == "samples" and not names_are_same:
-        raise ValueError(
-            "Sample names are not the same! Joining along samples with different "
-            "sample names will loose information and is not supported."
+    if not unsafe:
+        for ts_to_join in tensor_maps[1:]:
+            _check_maps(tensor_maps[0], ts_to_join, "join")
+
+        # Deduce if sample/property names are the same in all tensor_maps.
+        # If this is not the case we have to change unify the corresponding labels
+        # later.
+        if axis == "samples":
+            names_list = [tensor.sample_names for tensor in tensor_maps]
+        else:
+            names_list = [tensor.property_names for tensor in tensor_maps]
+
+        # We use functools to flatten a list of sublists::
+        #
+        #   [('a', 'b', 'c'), ('a', 'b')] -> ['a', 'b', 'c', 'a', 'b']
+        #
+        # A nested list with sublist of different shapes can not be handled by
+        # np.unique.
+        unique_names = np.unique(functools.reduce(operator.concat, names_list))
+
+        # Label names are unique: We can do an equal check only checking the lengths.
+        names_are_same = np.all(
+            len(unique_names) == np.array([len(names) for names in names_list])
         )
 
+        # It's fine to loose metadata on the property axis, less so on the sample axis!
+        if axis == "samples" and not names_are_same:
+            raise ValueError(
+                "Sample names are not the same! Joining along samples with different "
+                "sample names will loose information and is not supported."
+            )
```

Comment:
These are no tests but are required to construct the new names.

Reply:
yes i know, the idea here is to avoid assuming that the naming scheme. i do not know if it saves something